### PR TITLE
feat: report profile

### PR DESCRIPTION
**Pull request by spantheslayer**
## What does this PR do?
Implement a WIP feature: Reporting to profile

## Related issues
Modal is not working as expected.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Explanation of the changes
NA

### CODE REVIEW

File: apps/web/src/store/modals.ts [hunk 1]
```diff
@@ -42,5 +44,8 @@ export const useGlobalModalStateStore = create<GlobalModalState>((set) => ({
   showMobileDrawer: false,
   setShowMobileDrawer: (showMobileDrawer) => set(() => ({ showMobileDrawer })),
   showInvitesModal: false,
-  setShowInvitesModal: (showInvitesModal) => set(() => ({ showInvitesModal }))
+  setShowInvitesModal: (showInvitesModal) => set(() => ({ showInvitesModal })),
+  reportProfileModal: false,
```
Comment (bigint) [inline apps/web/src/store/modals.ts:48]: Rename it to `showReportProfileModal`

File: apps/web/src/components/Shared/ReportProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          name="reportReason"
+        />
+        <Radio
+          title="Frequently Posts Unwanted Content"
+          message="Spam; excessive mentions or replies"
+          name="reportReason"
+        />
+      </div>
+      <div className="mt-4 flex justify-center">
+        <Button type="submit" onClick={() => alert('gm 🚀')}>
```
Comment (bigint) [inline apps/web/src/components/Shared/ReportProfile.tsx:20]: We need the user profile ID here to be reported

File: apps/web/src/components/Shared/ReportProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      <div>
+        <Radio
+          title="Misleading Account"
+          message="Impersonation or false claims about identity or affiliation"
+          name="reportReason"
+        />
+        <Radio
+          title="Frequently Posts Unwanted Content"
+          message="Spam; excessive mentions or replies"
+          name="reportReason"
```
Comment (bigint) [inline apps/web/src/components/Shared/ReportProfile.tsx:16]: make sure to wrap strings under t`` macro for i18n

File: apps/web/src/components/Shared/ReportProfile.tsx (hunk 2 above)
Comment (bigint) [inline apps/web/src/components/Shared/ReportProfile.tsx:20]: Make the button full width

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by bigint** [comment]
Also add custom message textarea and try to wrap everything in zod instead of local states 🙇🏼

**Comment by bigint** [comment]
Also make the option only for logged in users, un authed users can't report

**Comment by bigint** [comment]
Run `pnpm i18n:extract`

**Comment by bigint** [comment]
padding is bit off, make sure the radio and title align the same, in most cases `p-5` works

Group 30

**Comment by bigint** [comment]
image

add lil more extra space between two items

File: apps/web/src/store/modals.ts (hunk 1 above)
Comment (bigint) [reply apps/web/src/store/modals.ts:48]: Still not resolved 🙇🏼

File: packages/ui/src/Radio.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import clsx from 'clsx';
+import type { FC, ReactNode } from 'react';
+
+interface RadioProps {
+  title?: string;
+  message?: ReactNode;
```
Comment (bigint) [inline packages/ui/src/Radio.tsx:6]: rename this to value instead of message

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/store/modals.ts (hunk 1 above)
Comment (spantheslayer) [reply apps/web/src/store/modals.ts:48]: I haven't pushed the code yet, I just market it for myself, pushing the change now

File: apps/web/src/components/Shared/Modal/ReportProfile/index.tsx [hunk 5]
```diff
@@ -4,32 +4,32 @@ import { t } from '@lingui/macro';
...
     schema: reportProfileSchema
   });
 
   return (
-    <div className="flex flex-col space-y-2 p-3">
+    <div className="flex flex-col space-y-2 p-5">
       <Form
         form={form}
         onSubmit={() => {
           alert('Submitted' + profile?.id);
```
Comment (bigint) [inline apps/web/src/components/Shared/Modal/ReportProfile/index.tsx:31]: @spantheslayer Can you also add what radio user is selected? 🙇🏼 

make sure to use enum here

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Shared/Modal/ReportProfile/index.tsx (hunk 5 above)
Comment (spantheslayer) [reply apps/web/src/components/Shared/Modal/ReportProfile/index.tsx]: when using padding of 5 , the allignment goes unexpected, thus I kept it at 3.

**Review by bigint — APPROVED** [review]
Code **LGTM** :shipit: 

LGTM

**Pull request merged**